### ISSUE TITLE
docker checkpoint create return error

### ISSUE THREAD

**Issue by elianka** [moby/moby#35235]
**Description**
i was trying to use docker checkpoint to demo container migration. meanwhile, i was failed to do this. 

**Steps to reproduce the issue:**
1.  set "experimental" as true in /etc/docker/daemon.json. and restart docker service.
2. start a container named looper

> docker run -d --name looper busybox:latest /bin/sh -c 'i=0; while true; do echo $i ; i=$(expr $i + 1); sleep 3; done'

3. use docker checkpoint to dump a checkpoint 

> docker checkpoint create looper checkpoint1

**Describe the results you received:**

> Error response from daemon: Cannot checkpoint container looper: rpc error: code = 2 desc = exit status 1: "criu failed: type NOTIFY errno 0\nlog file: /data/docker/containers/1c89315161e9d57c4d8a3e29e22210213d97c89bf57a16492556de71bf9a7f3e/checkpoints/checkpoint1/criu.work/dump.log\n"

![image](https://user-images.githubusercontent.com/4158830/31703492-604a6d8a-b3a2-11e7-9076-c8796a6e5579.png)

**Describe the results you expected:**
get checkpoint1 generated

**Additional information you deem important (e.g. issue happens only occasionally):**
the steps is following https://criu.org/Docker.

**Output of `docker version`:**

```
Client:
 Version:      1.13.1
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   092cba3
 Built:        Wed Feb  8 06:38:28 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.05.0-ce
 API version:  1.29 (minimum version 1.12)
 Go version:   go1.8.1
 Git commit:   89658be
 Built:        Thu Sep 28 09:50:28 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Output of `docker info`:**

```
Containers: 9
 Running: 1
 Paused: 0
 Stopped: 8
Images: 78
Server Version: 17.05.0-ce
Storage Driver: overlay
 Backing Filesystem: xfs
 Supports d_type: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: aa8187dbd3b7ad67d8e5e3a15115d3eef43a7ed1 (expected: 9048e5e50717ea4497b757314bad98ea3763c145)
runc version: 9df8b306d01f59d3a8029be411de015b7304dd8f (expected: 9c2d8d184e5da67c95d601382adf14862e4f2228)
init version: 949e6fa
Security Options:
 selinux
Kernel Version: 3.10.0-693.2.2.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 7.565 GiB
Name: thinkcenter
ID: CQFS:5YGK:U7DV:CCQI:HV46:2UQ6:VBIC:CSLK:4S4H:C3BT:TIIR:7VA3
Docker Root Dir: /data/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 21
 Goroutines: 27
 System Time: 2017-10-18T14:20:49.174085507+08:00
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 127.0.0.0/8
Registry Mirrors:
Live Restore Enabled: true
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
physical machine

**Comment by fntlnz** [comment]
I'm able to reproduce, this is under discussion also in criu https://github.com/checkpoint-restore/criu/issues/393

The problem is that containerd, mine is (06b9cb35161009dcb7123345749fef02f7cea8e0) doesn't pass the `empty-ns` option to criu at restore time but only at creation time.
This has been addressed in the latest versions of go-runc that now is used in the master of containerd. 
A vendor upgrade is needed if possible, I say if possible because containerd changed a lot recently.

**Comment by fntlnz** [comment]
@crosbymichael WDYT?

**Comment by avagin** [comment]
We workaround this issue in CRIU and it is going to be released in the CRIU 3.6, which is scheduled on the next Monday.

**Comment by fntlnz** [comment]
@avagin thanks for your feedback, @thajeztah please look at this

**Comment by yongtang** [comment]
As CRUI 3.6 as been merged in #35695, I think this issue could be closed. Please feel free to reopen if the issue still exists.
